lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java

```diff
@@ -0,0 +1,275 @@
+            userId);
+      }
+    } finally {
+      taskanaEngine.returnConnection();
+      LOGGER.debug("exit from updateTaskComment()");
+    }
+
+    return taskCommentImplToUpdate;
+  }
+
+  TaskComment createTaskComment(TaskComment taskCommentToCreate)
+      throws NotAuthorizedException, TaskNotFoundException, InvalidArgumentException {
+
+    LOGGER.debug("entry to setTaskComment (taskCommentToCreate = {})", taskCommentToCreate);
+
+    TaskCommentImpl taskCommentImplToCreate = (TaskCommentImpl) taskCommentToCreate;
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      taskService.getTask(taskCommentImplToCreate.getTaskId());
+
+      validateNoneExistingTaskCommentId(taskCommentImplToCreate.getId());
+
+      initDefaultTaskCommentValues(taskCommentImplToCreate);
+
+      taskCommentMapper.insert(taskCommentImplToCreate);
+
+    } finally {
+
+      taskanaEngine.returnConnection();
+
+      LOGGER.debug("exit from setTaskComment()");
+    }
+
+    return taskCommentImplToCreate;
+  }
+
+  void deleteTaskComment(String taskCommentId)
+      throws NotAuthorizedException, TaskCommentNotFoundException, TaskNotFoundException,
+          InvalidArgumentException {
+
+    LOGGER.debug("entry to deleteTaskComment (taskComment = {}", taskCommentId);
+
+    String userId = CurrentUserContext.getUserid();
+
+    validateExistingTaskCommentId(taskCommentId);
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      TaskComment taskCommentToDelete = getTaskComment(taskCommentId);
+
+      if (taskCommentToDelete != null) {
+
+        taskService.getTask(taskCommentToDelete.getTaskId());
+
+      } else {
+        throw new TaskCommentNotFoundException(
+            taskCommentId, String.format("The TaskComment with ID %s wasn't found", taskCommentId));
+      }
+
+      if (taskCommentToDelete.getCreator().equals(userId)) {
+
+        taskCommentMapper.delete(taskCommentId);
+
+        LOGGER.debug("taskComment {} deleted", taskCommentToDelete.getId());
+
+      } else {
+        throw new NotAuthorizedException(
+            NOT_AUTHORIZED + taskCommentToDelete.getCreator() + BUT_CURRENT_USER_IS + userId,
+            userId);
+      }
+
+    } finally {
+      taskanaEngine.returnConnection();
+      LOGGER.debug("exit from deleteTaskComment()");
+    }
+  }
+
+  List<TaskComment> getTaskComments(String taskId)
+      throws TaskCommentNotFoundException, NotAuthorizedException, TaskNotFoundException {
+
+    LOGGER.debug("entry to getTaskComments (taskId = {})", taskId);
+
+    List<TaskComment> result =
+        taskCommentMapper.findByTaskId(taskId).stream().collect(Collectors.toList());
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      taskService.getTask(taskId);
+
+      if (result == null || result.isEmpty()) {
+        throw new TaskCommentNotFoundException(
+            taskId, "TaskComments for TaskId " + taskId + " were not found");
+      }
```

Comment:
You should return an empty list in this case, because this is no exceptional situation.
In other places of the product there is also returned an empty list in such cases.
Refer e.g. to queries, or to TaskService.updateTasks(..)

Reply:
done